scan/tls_handshake.go

```diff
@@ -30,8 +54,9 @@ func sayHello(host string, ciphers []uint16, vers uint16) (cipherIndex int, err
 	config.MinVersion = vers
 	config.MaxVersion = vers
 	config.CipherSuites = ciphers
+	config.CurvePreferences = curves
 	conn := tls.Client(tcpConn, config)
-	serverCipher, serverVersion, err := conn.SayHello()
+	serverCipher, serverCurveType, serverCurve, serverVersion, err := conn.SayHello()
 	conn.Close()
 	if err != nil {
 		return
```

Comment:
This will shadow all other possibly legitimate errors possibly returned by conn.SayHello, no? Or are you saying that there aren't any legitimate errors that can happen there (or that the loop in `cipherSuiteScan` should break in that case regardless)? I guess one could put the sentinel value into the cf-tls helper files and use reflect.DeepEqual to see how to set it, such that other errors are still passed up to here.